pkg/user/repository.go

```diff
@@ -41,6 +35,12 @@ func (r repository) create(u *model.User) error {
 func (r repository) findByEmail(email string) (*model.User, error) {
 	var u *model.User
 	err := r.db.Where("email = ?", email).First(&u).Error
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			err := fmt.Errorf("failed to find user with email %s: %v", email, err)
```

Comment:
I would default to use `%q` for strings (email in that case). Quoting the string in the error/logs helps to see any issues with whitespace added into a string.